source/development/feature-flags.html.md

```diff
@@ -0,0 +1,76 @@
+
+### Try to minimise the scope of the change
+Try to have your flag controlling as little as possible. Minimising how much is behind a feature flag reduces the burden of technical management and the impact of the change itself. Check if there’s anything in your change that can be released outside the feature flag as a separate change.
+
+### Be aware of the need for multiple feature flags
+Your feature may impact multiple parts of our tech stack eg: the Sass and the javascript. In which case you should ensure that your flag is built in such a way that the impacted element states are “synced” where possible. For example, if you have a flag in a component macro that controls the markup of that macro but also requires a Sass change, your flag should automatically add a CSS class that applies that styling change.
+
+## The lifecycle of a feature flag
+Follow these instructions for applying, working with and finally getting rid of feature flags.
+
+### Implementing a feature flag
+Feature flags and the change they’re controlling should be released in a minor release aka: X.X.X. As part of this release, automated tests should be written to account for govuk-frontend with the feature flag on and off to check for unforeseen impact to parts of the codebase that the flag wasn’t intended to affect.
+
+At this stage, we should create github issues to remove the feature flag in a future major release and apply them to ourplanning accordingly.
+
+We communicate in release notes and guidance what this feature flag impacts, how to use it and how “experimental” the new change is.
+
+### Working with a feature flag in flight
+By turning a feature flag on, we presume a user is opting into the following conditions:
+
+We may release “breaking” changes to things behind the feature flag. We will communicate this in release notes.
+Depending on the change itself and if it’s experimental or not, the thing behind the feature flag may not be 100% operable ie: there may be bugs, users may provide negative feedback etc.
+
+It’s important for us to monitor community feedback on the feature flag, its use and the change itself so that we can iterate the change before it’s removal.
+
+### Removing a feature flag
+Feature flags should be removed in major versions of govuk-frontend and their removal should be considered breaking. Ideally we should plan for removal of feature flags in the next major release after the feature flag is implemented eg: if we release a feature flag in 5.2.0, we should remove it in 6.0.0. There are numerous reasons why we might not always be able to do this however, such as users commenting that they need more time whilst we have other pressing breaking changes we want to make. Therefore, if we can’t remove the feature flag in the following major release, we should decide per future breaking release if the feature flag can be removed.
```

Comment:
```suggestion
You should always aim to remove feature flags in the next major version of govuk-frontend after the flag was implemented and consider their removal as a breaking change. For example, if you release a feature flag in 5.2.0, you should remove it in 6.0.0. There are many reasons why this might not always be possible, such as users asking for more time or other breaking changes taking priority. If you cannot complete the removal in the next major release, you should plan to remove the feature flag in a future breaking release.
```

Reply:
Let me know if you want to meet to speed up the editing process.

Reply:
How about starting the section with something like:
'You should maintain a consistent process where you check for active feature flags and decide if you still need them.'

Reply:
Implemented 😎